app/models/dialog.rb

```diff
@@ -26,6 +26,14 @@ def self.seed
     end
   end
 
+  def readonly?
+    return true if super
+    resource_actions.each do |action|
+      return true if action.resource.try(:readonly?)
```

Comment:
@bzwei Can you call the `readonly?`method on the `resource_action` object instead of calling it on `action.resource`?